sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoin.scala

```diff
@@ -55,11 +56,43 @@ case class ShuffledHashJoin(
   override def requiredChildDistribution: Seq[Distribution] =
     ClusteredDistribution(leftKeys) :: ClusteredDistribution(rightKeys) :: Nil
 
+  private def buildHashedRelation(iter: Iterator[UnsafeRow]): HashedRelation = {
+    // try to acquire some memory for the hash table, it could trigger other operator to free some
+    // memory. The memory acquired here will mostly be used until the end of task.
+    val context = TaskContext.get()
+    val memoryManager = context.taskMemoryManager()
+    var acquired = 0L
+    var used = 0L
+    context.addTaskCompletionListener((t: TaskContext) =>
+      memoryManager.releaseExecutionMemory(acquired, MemoryMode.ON_HEAP, null)
+    )
+
+    val copiedIter = iter.map { row =>
+      // It's hard to guess what's exactly memory will be used, we have a raft guess here.
+      // TODO: use BytesToBytesMap instead of HashMap for memory efficiency
+      val needed = 150 + row.getSizeInBytes
```

Comment:
Can you please explain (and possibly comment in code) the reason behind choosing 150?